app/controllers/api/widget_sets_controller.rb

```diff
@@ -0,0 +1,59 @@
+module Api
+  class WidgetSetsController < BaseController
+    REQUIRED_FIELDS_TO_COPY = %w[name].freeze
+    ALLOWED_FIELDS = REQUIRED_FIELDS_TO_COPY + %w[group description guid read_only set_data].freeze
```

Comment:
with 'read_only' here, does this mean the API can create read_only widget sets and cannot delete them ?  I usually think of read_only as set to true for some system seeded resources.